mirgecom/boundary.py

```diff
@@ -51,74 +60,379 @@ class FluidBoundary(metaclass=ABCMeta):
+            :func:`~mirgecom.viscous.viscous_facial_flux_central`.
+
+        Returns
+        -------
+        :class:`mirgecom.fluid.ConservedVars`
+        """
+
+    @abstractmethod
+    def cv_gradient_flux(self, discr, btag, gas_model, state_minus, **kwargs):
+        """Get the boundary flux for the gradient of the fluid conserved variables.
+
+        This routine returns the facial flux used by the gradient operator to
+        compute the gradient of the fluid solution on a domain boundary.
+
+        Parameters
+        ----------
+        discr: :class:`~grudge.eager.EagerDGDiscretization`
+
+            A discretization collection encapsulating the DG elements
+
+        state_minus: :class:`~mirgecom.gas_model.FluidState`
+
+            Fluid state object with the conserved state, and dependent
+            quantities for the (-) side of the boundary specified by
+            *btag*.
+
+        gas_model: :class:`~mirgecom.gas_model.GasModel`
+
+            Physical gas model including equation of state, transport,
+            and kinetic properties as required by fluid state
+
+        Returns
+        -------
+        :class:`mirgecom.fluid.ConservedVars`
+        """
+
+    @abstractmethod
+    def temperature_gradient_flux(self, discr, btag, gas_model, state_minus,
+                                  **kwargs):
+        """Get the boundary flux for the gradient of the fluid temperature.
+
+        This method returns the boundary flux to be used by the gradient
+        operator when computing the gradient of the fluid temperature at a
+        domain boundary.
+
+        Parameters
+        ----------
+        discr: :class:`~grudge.eager.EagerDGDiscretization`
+
+            A discretization collection encapsulating the DG elements
+
+        state_minus: :class:`~mirgecom.gas_model.FluidState`
+
+            Fluid state object with the conserved state, and dependent
+            quantities for the (-) side of the boundary specified by
+            *btag*.
+
+        gas_model: :class:`~mirgecom.gas_model.GasModel`
 
+            Physical gas model including equation of state, transport,
+            and kinetic properties as required by fluid state
 
+        Returns
+        -------
+        numpy.ndarray
+        """
+
+
+# This class is a FluidBoundary that provides default implementations of
+# the abstract methods in FluidBoundary. This class will be eliminated
+# by resolution of https://github.com/illinois-ceesd/mirgecom/issues/576.
+# TODO: Don't do this. Make every boundary condition implement its own
+# version of the FluidBoundary methods.
 class PrescribedFluidBoundary(FluidBoundary):
     r"""Abstract interface to a prescribed fluid boundary treatment.
 
     .. automethod:: __init__
     .. automethod:: inviscid_divergence_flux
+    .. automethod:: cv_gradient_flux
+    .. automethod:: viscous_divergence_flux
+    .. automethod:: temperature_gradient_flux
     """
 
     def __init__(self,
```

Comment:
This should probably be converted into a proper numpy docstring with these comments: https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring

Reply:
Can we escalate this to an issue and discuss class doc convention with the group?   I am happy to convert the docstring for this class, but all the classes in mirgecom are currently following the same (non)convention as this one.

Reply:
Yep good by me.

Reply:
#673